src/editor.js

```diff
@@ -23,7 +23,12 @@ class MonacoEditor extends React.Component {
       // Consider the situation of rendering 1+ times before the editor mounted
       if (this.editor) {
         this.__prevent_trigger_change_event = true;
-        this.editor.setValue(this.__current_value);
+        this.editor.pushUndoStop();
+        this.editor.executeEdits('', [{
+          range: this.editor.getModel().getFullModelRange(),
+          text: this.__current_value
+        }], [new monaco.Range(1, 1, 1, 1)]);
+        this.editor.pushUndoStop();
```

Comment:
What's this undo stop and why do we need to push two?

Reply:
Each undo will revert to the previous "major" change, grouping intermediate "minor" edits together as one single change. I consider external value change to be a major change, so I push an undo stop to make it become a major change.

Not sure if pushing undo stops twice is the correct way to do it, but that's how it's used in [VSCode](https://github.com/microsoft/vscode/blob/2b518c120d9422469cfc5bc63c4abee5355915b8/src/vs/editor/contrib/linesOperations/linesOperations.ts#L48), so I just follow them.